### ISSUE TITLE
[Bug]: cleanup local groups not working for income categories

### ISSUE THREAD

**Issue by yauyenching** [actualbudget/actual#3425]
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

Hello, when using income categories as sources for End of Month cleanup, I find that it budgets all income funds, rather than just the income categories I defined with `#cleanup source`. 

Steps to reproduce

1. Have remaining funds to be budgeted across different income categories
1. Define one income category with `#cleanup source`
2. Define another budget category with `#cleanup sink`
3. Run End of Month cleanup
4. Observe all remaining funds budgeted disappeared, not just funds from income category with `#cleanup source`

Thanks so much in advance!

### Where are you hosting Actual?

Pikapods

### What browsers are you seeing the problem on?

Firefox

### Operating System

Windows 11

**Comment by youngcw** [comment]
Im confused on why you would be trying to use the cleanup script for income categories.  Those categories don't use "to budget" funds.

Are you using the tracking budget?  If so there isn't much reason to use the cleanup script in the first place since the cleanup is meant to approximate the tracking budget inside an envelope budget.

**Comment by youngcw** [comment]
Lacking info.  Closing

**Comment by yauyenching** [comment]
@youngcw hi! here to provide more info on my use case! i often get interest/cashback/misc income credit in the middle/end of the month, and i'd love to automate allocating whatever in these categories into my savings + vacation fund. i realized i could do the same with templates -- but using "#template 50% of Cashback" x3 is tedious (3 different income categories: Interest, Cashback, Income (Misc)) and using "#template remainder" also wouldn't work for me considering that I like to manually allocate the majority of my main income.

**Comment by youngcw** [comment]
Have you looked into the end of month cleanup?  It probably could do this for you all at once at the end of the month

**Comment by yauyenching** [comment]
yes i'd like to use end of month cleanup! but like I mentioned in the above bug, it doesn't work very well for income categories. i tried it again just now, and it cleaned up from ALL my income categories (including my 'Paycheck' category that I had yet to budget the funds from and does not have `#cleanup source` defined, rather than just the 3 income categories i had written `#cleanup source` in.

**Comment by yauyenching** [comment]
ah, so you're saying i should budget all my main income first and then let `#cleanup sink` do as desired? makes sense! it may not work the best for my use case since i tend to get my Paycheck and savings account interest around the same time... I might stick to `#template 50% of ...` -- but i greatly appreciate you engaging with me on this! 😄

**Comment by youngcw** [comment]
The templates work really well if you are using the month ahead process.  I use the `#template X% of prev <income cat>` templates for things and other general templates.  That way I know exactly the income Im working with right from the get go.

That said, they don't work as well if you have extra funds in the middle of the month.  Using template priorities can help where you can re-run and build up budgets as funds are available.

**Comment by yauyenching** [comment]
that's a great read! 💯 thanks for linking it -- and again really appreciate your responsiveness on my various issues (lol)! migrated from YNAB4 and i really love its open-source nature with the capacity for complex budgets. looking forward to maybe contributing someday! 🤗
